web/src/main/java/com/microsoft/applicationinsights/internal/config/WebReflectionUtils.java

```diff
@@ -0,0 +1,21 @@
+package com.microsoft.applicationinsights.internal.config;
+
+import static com.microsoft.applicationinsights.internal.config.ReflectionUtils.addClass;
+
+public class WebReflectionUtils {
+    public static void initialize() {

```

Comment:
This is called in `<clinit>` of the Filter. It should add all these classes so they are initialized properly.

Tests for this are pending.

Reply:
I am suspecting this scenario can break springboot when we add filter and it's initialize method is called. Here is the reason : SpringBoot adds the beans for all these modules using DI. That happens when the framework initializes and it also adds the Filter as a a bean. Now when this init method is called it will try to add all this things back into the class path when they are already present which can cause potential conflicts. We need to test this with Springboot app before merging.

Reply:
I didn't follow what you meant here. This will be called before the initialize method in WebRequestTrackingFilter. This doesn't add anything to the classpath. If this is "already on the classpath" it's not causing conflicts, it's referencing what's already there.

Reply:
sounds good. We should definitely test it though before shipping with a SpringBootApp to make sure that it works smoothly.